File: Python/7py 2/letter_removal.py
"""
|--------------------------------------------------------------------------
| Simple letter removal
|--------------------------------------------------------------------------
|
"""

def solve(st, k):
   r = sorted(list(st))[:k]
   for val in r:
      st = st.replace(val, "", 1)
   return st


solve= lambda s, k: [s:=s.replace(l,'',1) for l in sorted(s)[:k]][-1] if k else s


def solve(st,k): 
   for x in range(26):
      while k and chr(x+97) in st:
         st = st.replace(chr(x+97), "", 1)
         k -= 1
   return st


print(solve('abracadabra', 1),'bracadabra')
print(solve('abracadabra', 2),'brcadabra')
print(solve('abracadabra', 6),'rcdbr')
print(solve('abracadabra', 8),'rdr')
print(solve('abracadabra', 50),'')

File: PHP/7php 2/letter_removal.php
<?php
/*
|--------------------------------------------------------------------------
| Simple letter removal
|--------------------------------------------------------------------------
|
*/

function solve($s, $k) {
   $alphabet = range('a', 'z');
   $alphabetCount = count($alphabet);
   for ($i = 0; $i < $alphabetCount; $i++) {
      $val = $alphabet[$i];
      $count = 0;
      $s = preg_replace("/$val/", '', $s, $k, $count);
      $k -= $count;
   }
   return $s;
}


function solve($word, $step, $start='a'){
	return $step > 0 && !empty($word) ? solve(preg_replace("/$start/i", '', $word, $step, $count), $step-$count, chr(ord($start)+1)) : $word;
}


function solve($s, $k) {
   foreach (range('a', 'z') as $val) {
      $s = preg_replace("/$val/", '', $s, $k, $count);
      $k -= $count;
   }
   return $s;
}


print_r(solve("abracadabra", 1) . " bracadabra
");
print_r(solve("abracadabra", 2) . " brcadabra
");
print_r(solve("abracadabra", 6) . " rcdbr
");
print_r(solve("abracadabra", 8) . " rdr
");
print_r(solve("abracadabra", 50) . " 
");